packages/opentelemetry-instrumentation-grpc/src/instrumentation.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+} from '@opentelemetry/instrumentation';
+import { GrpcInstrumentationConfig } from './types';
+import { VERSION } from './version';
+import { getGrpcPatches } from './grpc';
+import { getGrpcJsPatches } from './grpc-js';
+
+/** The metadata key under which span context is stored as a binary value. */
+export const GRPC_TRACE_KEY = 'grpc-trace-bin';
+
+export class GrpcInstrumentation extends InstrumentationBase {
+  constructor(
+    protected _config: GrpcInstrumentationConfig & InstrumentationConfig = {}
+  ) {
+    super('@opentelemetry/instrumentation-grpc', VERSION, _config);
+  }
+
+  public setConfig(
+    config: GrpcInstrumentationConfig & InstrumentationConfig = {}
+  ) {
+    this._config = Object.assign({}, config);
+  }
+
+  /**
+   * @internal
+   * Public reference to the protected BaseInstrumentation shimmer utils to be used by this
+   * plugin's external helper functions
+   */
+  public getShimmer() {
```

Comment:
All those methods `getShimmer`, `getLogger`, `getTracer`, and `getConfig` basically volatile the whole idea of protected internals. Because many functionalities are spread across different utils class instead of passing instrumentation class as dependency injection directly and then trying to use all of the internals I think it would be safer if you simply pass (dependency injection) only the functions you need in certain functionality instead of whole class. This way you will be able to still use it without exposing the internals using extra public functions. Otherwise the exposed internals can be also used by any 3rd parties which I think should not be possible. WDYT ?